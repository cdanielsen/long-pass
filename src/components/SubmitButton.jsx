import React from 'react'
import Button from 'material-ui/Button'
import Lock from 'material-ui-icons/Lock'

import PropTypes from 'prop-types'

function SubmitButton({ id, clickHandler }) {
  return (
    <Button fab color="primary" id={id} onClick={clickHandler}>
      <Lock style={{ width: 40, height: 40 }} />
    </Button>
  )
}

SubmitButton.propTypes = {
  clickHandler: PropTypes.func.isRequired,
  id: PropTypes.string,
}

export default SubmitButton
