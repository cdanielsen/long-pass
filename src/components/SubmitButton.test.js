import React from 'react'
import SubmitButton from './SubmitButton.jsx'
import { shallow } from 'enzyme'

describe('SubmitButton component', () => {
  const props = {
    id: 'lolbutton',
    clickHandler: jest.fn()
  }
  const wrapper = shallow(<SubmitButton {...props} />)

  test('Given props, should render with the correct attributes', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.prop('id')).toEqual(props.id)
    expect(wrapper.prop('onClick')).toEqual(props.clickHandler)
  })

  test('When clicked, should call event handler', () => {
    wrapper.simulate('click')
    expect(props.clickHandler.mock.calls.length).toEqual(1)
  })
})
