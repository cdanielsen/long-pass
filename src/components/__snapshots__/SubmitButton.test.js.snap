// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton component Given props, should render with the correct attributes 1`] = `
<withStyles(Button)
  color="primary"
  fab={true}
  id="lolbutton"
  onClick={[Function]}
>
  <pure(Lock)
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  />
</withStyles(Button)>
`;
